src/internal/client.ts

```diff
@@ -864,4 +865,12 @@ export class TypedClient {
     const res = await this.makeRequestAsync({ method, bucketName, objectName, query })
     return xmlParsers.parseListParts(await readAsString(res))
   }
+
+  listBuckets(callback: ResultCallback<BucketItemFromList[]>): Promise<BucketItemFromList[]>
```

Comment:
This maybe convenient but it doesn't help TypeScript to narrow overload, should split this with a promise api and a callback api

Reply:
Could you please elaborate this? is the overload not sufficient ?

Reply:
`listBuckets(()=>{}).then(result => {}, err => {} )` is not actually valid, but still pass type checking

Reply:
And the easiest way is always to bump major version and drop callback style api...

Reply:
Yes. Thank you. :+1: